### PR TITLE
Adds files module and get function

### PR DESCRIPTION
**Pull request by rob5standingby**
## Why?
We need a `Files` module and a `get` function to retrieve some metadata about a file from the new Files API!

## JIRA
https://ndustrialio.atlassian.net/browse/CFILE-2 📖 

## What changed?
- Added `Files` module. Based the File typedef on [this](https://contxt.readme.io/reference#files-retrieve-a-file).
- Added factory for `file`
- Added tests

### CODE REVIEW

File: src/files/index.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+   *
+   * @param {string} fileId The ID of the file
+   *
+   * @returns {Promise}
+   * @fulfill {File} Information about a file
+   * @reject {Error}
+   *
+   * @example
+   * contxtSdk.files
+   *   .get(12345)
```
Comment (leeallen337) [inline src/files/index.js]: Aren't the id's UUIDs?

File: src/files/index.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+   *   .then((file) => console.log(file))
+   *   .catch((err) => console.log(err));
+   */
+  get(fileId) {
+    if (!fileId) {
+      return Promise.reject(
+        new Error('A file ID is required for getting information about a file')
```
Comment (leeallen337) [inline src/files/index.js]: Do we put periods at the end of our error messages or no?

File: src/files/index.js (hunk 1 above)
Comment (leeallen337) [reply src/files/index.js]: https://github.com/ndustrialio/files-api/blob/master/models/file.js#L12

File: support/fixtures/factories/file.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+factory
+  .define('file')
+  .option('fromServer', false)
+  .attrs({
+    createdAt: () => faker.date.past().toISOString(),
+    contentType: () => faker.system.mimeType(),
+    description: () => faker.hacker.phrase(),
+    filename: () => `${faker.hacker.noun()}.${faker.system.commonFileExt()}`,
+    id: () => faker.random.number(),
```
Comment (leeallen337) [inline support/fixtures/factories/file.js]: Looks like this should be a UUID.
https://github.com/ndustrialio/files-api/blob/master/models/file.js#L12

File: src/files/index.js (hunk 1 above)
Comment (rob5standingby) [reply src/files/index.js]: Greg and I are out of sync. I documented them as Integers ;P

File: src/files/index.js (hunk 2 above)
Comment (rob5standingby) [reply src/files/index.js]: No, it doesn't appear so based on some of the other modules

File: src/config/audiences.js [hunk 4]
```diff
@@ -49,6 +49,16 @@ export default {
...
     }
   },
+  files: {
+    production: {
+      clientId: '6jPAqJCMFMRP5t7Rw2X19qt2EtTfdQ2Q',
+      host: 'https://files.api.ndustrial.io'
+    },
+    staging: {
+      clientId: '6jPAqJCMFMRP5t7Rw2X19qt2EtTfdQ2Q',
+      host: 'https://files-staging.api.ndustrial.io'
```
Comment (greg5green) [inline src/config/audiences.js]: These client IDs are different now. And might change again in the future for reasons.

Current:
Staging: `SLE310LY4nDT5p06sUGPVOLlGXk5osbQ`
Production: `VZJ2MRd71dM9oX2hB1EVAEjUe6loL7pl`

File: src/files/index.spec.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        expectedFileId = faker.random.uuid();
+        fileFromServerAfterFormat = fixture.build('file', {
+          id: expectedFileId
+        });
+
+        fileFromServerBeforeFormat = fixture.build(
+          'file',
+          { id: expectedFileId },
+          { fromServer: true }
+        );
```
Comment (greg5green) [inline src/files/index.spec.js]: What do you think about doing this a little differently?

```
fileFromServerAfterFormat = fixture.build('file');
fileFromServerBeforeFormat = fixture.build('file', fileFromServerAfterFormat, { fromServer: true });
```

File: support/fixtures/factories/file.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+const faker = require('faker');
+
+factory
+  .define('file')
+  .option('fromServer', false)
+  .attrs({
+    createdAt: () => faker.date.past().toISOString(),
+    contentType: () => faker.system.mimeType(),
+    description: () => faker.hacker.phrase(),
+    filename: () => `${faker.hacker.noun()}.${faker.system.commonFileExt()}`,
```
Comment (greg5green) [inline support/fixtures/factories/file.js]: What about `() => faker.system.commonFileName()`?

Comment (rob5standingby) [reply support/fixtures/factories/file.js]: Sure, wasn't aware of that one!

File: src/files/index.spec.js (hunk 5 above)
Comment (rob5standingby) [reply src/files/index.spec.js]: Ah, I keep forgetting about that option.

File: src/config/audiences.js (hunk 4 above)
Comment (rob5standingby) [reply src/config/audiences.js]: Ah, thanks. I'll update them to those for now.

**Pull request merged**